cmd/container-disk-v2alpha/main.c

```diff
@@ -0,0 +1,204 @@
+#include <errno.h>
+#include <getopt.h>
+#include <libgen.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+#include <signal.h>
+#include <fcntl.h>
+
+#include <time.h>
+
+#define LISTEN_BACKLOG 50
+#define READINESS_PROBE_FILE "/healthy"
+
+char copy_path[108];
+
+void error_log(const char *format, ...)
+{
+    va_list arglist;
+
+    time_t ltime; /* calendar time */
+    ltime=time(NULL); /* get current cal time */
+    fprintf(stderr, "%s",asctime(localtime(&ltime)));
+    fprintf(stderr, "error: ");
+    va_start(arglist, format);
+    fprintf(stderr, format, arglist);
+    va_end(arglist);
+}
+
+void sig_handler(int signo) {
+    if (copy_path != NULL) {
+        unlink(copy_path);
+    }
+    exit(0);
+}
+
+static void *socket_check(int fd, void *arg) {
+    struct stat st = {0};
+    char *copy_path;
+    copy_path = (char *)arg;
+    int connfd;
+    bool connReceived = false;
+
+    /*
+     * Periodically check the following:
+     *
+     * First, if the socket file still exists. We use it as an indicator to
+     * shut down the container, in case that we don't receive a signal from
+     * kuberenetes. We had issues with receiving the signal in time on
+     * different container runtime implementations over time and therefore use
+     * this as a precaution.
+     *
+     * Second accept socket connections to avoid filling up the SYN queue. If
+     * ther is a connection, we close it immediatley and immediately try to
+     * read the next connection until there are no more connections in the
+     * queue. Once the queue is empty, we fall back to sleep for a second.
+     *
+     * If within that second more than 50 connections enter the queue, then we
+     * clearly have a bug in virt-handler.
+     */
+    for (;;) {
+        if (!connReceived) {
+            sleep(1);
```

Comment:
what's the purpose of the sleep?  Is it to keep this loop from spinning because accept is occurring on a non-blocking socket?

if so, lets use a select or poll on the fd to only call accept when there's something to process

Reply:
afaik poll only allows me to wait on multiple file descriptors, but still blocks. We also want to check if the socket file disappeared, since we use it as a further shutdown signal for the container-disk. I don't know if you remember it but we had on many occasions forever terminating pods, because the `compute` container was terminated but the container-disk pod never received the signal. This additional check allows to transmit the shutdown signal also via virt-handler in ddition.

Reply:
oh yeah, i forgot about that. makes sense, and that's something I'll have to remember to add back to my hostpath PR as well. I removed the removal of the socket file.